### ISSUE TITLE
FF 93 has removed KeyboardEvent.initKeyEvent: "Synthetic keyboard events are not supported in this browser"

### ISSUE THREAD

**Issue by niloc132** [gwtproject/gwt#9742]
**GWT version:** GWT 2.9.0, latest HEAD
**Browser (with version):** Firefox 93+
**Operating System:** Any

---

##### Description
According to https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/initKeyEvent, initKeyEvent has been disabled, but a preference still exists to reenable it, and will be completely removed soon.

com.google.gwt.dom.client.DOMImplMozilla.createKeyEventImpl currently uses this method, but tests for its existence. In order to support IE11, if the method doesn't exist, the following console message will be logged:
```
Synthetic keyboard events are not supported in this browser
```

The proper fix is to use KeyboardEvent which will certainly work in any recent version of Firefox (though, [according to MDN, not IE11](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/KeyboardEvent)).

##### Steps to reproduce
Run GWT's TextCellTest.testOnBrowserEventKeyUp, or otherwise attempt to use `Document.get().createKeyUpEvent` and the other keyboard event method in firefox 93+. 

##### Known workarounds
It is possible to provide a custom DOMImpl which correctly implements this method, subclassing DOMImplMozilla, and then rebinding DOMImpl when the `user.agent` selection property is `gecko1_8`. Then, all keyboard events can be created correctly with the KeyboardEvent constructor in js. Note that this is how https://github.com/gwtproject/gwt-dom has implemented this feature.

**Comment by niloc132** [comment]
Draft patch, solves at least the cell tests, currently running all tests. https://github.com/niloc132/gwt/commit/1f18f63e180647c87d6cabaee1f5955844a6489c

**Comment by niloc132** [comment]
I misread the closure externs and mdn for the init object, but with that fixed for a real browser, my fix cannot work in general - HtmlUnit 2.19 doesn't correctly support the KeyboardEvent init object fields, so calling initKeyEvent is still necessary. To support htmlunit-2.19-as-firefox, one would need to either use the old document.createElement+initKeyEvent even if KeyboardEvent exists, or call initKeyEvent after using ostensibly the working constructor. Either way though, that means adding a branch to production code just to handle old HtmlUnit.

Instead, I'm going to roll this fix forward into the htmlunit update, so that recent FF works properly, and htmlunit correctly will simulate the browser we're avoiding actually running...

**Comment by niloc132** [comment]
GWT 2.10 will be released in the next few days, marking this as closed.
